### PR TITLE
Switch meta replication from sync to replicate

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Since the core problem is that we don't immediately catch promise rejections, and this only happens because of the implementation of `PouchDB.sync`, I'm switching meta database replication to instead use `replicate.from` and `replicate.to`, which behave as `zone.js` expects. 

medic/cht-core#7034

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
hey @garethbowen . Would you mind taking a look? Thanks for your time!

File: webapp/src/ts/services/db-sync.service.ts [hunk 1]
```diff
@@ -202,11 +202,15 @@ export class DBSyncService {
...
       .then(info => currentSeq = info.update_seq)
-      .then(() => {
-        return local
-          .sync(remote)
-          .on('complete', (info) => telemetryEntry.recordSuccess(info))
-          .on('error', (err) => telemetryEntry.recordFailure(err, this.knownOnlineState));
+      .then(() => Promise.all([
+        local.replicate.to(remote),
+        local.replicate.from(remote),
+      ]))
```
Comment (garethbowen) [inline webapp/src/ts/services/db-sync.service.ts:208]: I'm in two minds about whether this should be in series or parallel. I think I prefer running in series to keep the background thread light. Why do you prefer running in parallel?

File: tests/e2e/contacts/new-lineage-structure.wdio-spec.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ describe('Create new lineage structure', () => {
   });
 
   after(async () => {
-    await utils.revertDb([], true);
+    await utils.revertDb([], 'api');
```
Comment (garethbowen) [inline tests/e2e/contacts/new-lineage-structure.wdio-spec.js:24]: If I'm looking at the right utils, the second parameter is always treated as a boolean: https://github.com/medic/cht-core/blob/8bdce5bdc3731814e06aa113154ae1792b5a2323/tests/utils.js#L258

What am I missing?

**Review by garethbowen — CHANGES_REQUESTED** [review]
Lovely test as always. Two questions inline...

File: webapp/src/ts/services/db-sync.service.ts (hunk 1 above)
Comment (dianabarsan) [reply webapp/src/ts/services/db-sync.service.ts:208]: I just kept it as `sync` does it, which starts both in parallel. 
I have no preference one way or the other.

File: tests/e2e/contacts/new-lineage-structure.wdio-spec.js (hunk 2 above)
Comment (dianabarsan) [reply tests/e2e/contacts/new-lineage-structure.wdio-spec.js:24]: You're right! I'll revert. Only `updateSettings` passes the parameter along. I'll revert. Thanks!

File: webapp/src/ts/services/db-sync.service.ts (hunk 1 above)
Comment (garethbowen) [reply webapp/src/ts/services/db-sync.service.ts:208]: Ok. Consistency makes sense. Good to go!

**Review by garethbowen — APPROVED** [review]
Approved pending revert of `revertDb` args.

**Pull request merged**